subsys/net/ip/icmpv4.c

```diff
@@ -174,7 +259,10 @@ int net_icmpv4_send_error(struct net_pkt *orig, u8_t type, u8_t code)
 	int err = -EIO;
 
 	if (NET_IPV4_HDR(orig)->proto == IPPROTO_ICMP) {
-		if (NET_ICMP_HDR(orig)->code < 8) {
+		struct net_icmp_hdr icmp_hdr[1];
+
+		if (!net_icmpv4_get_hdr(orig, icmp_hdr) ||
+		    icmp_hdr->code < 8) {
```

Comment:
Well, that's quite a creative hack of defining a struct var as as array of size 1 and then use -> to access its fields - never using & operator! I don't believe I ever saw this, so I won't even ask why it's used here, as they say, live and learn ;-).

However, isn't it an API contract for net_icmpv4_get_hdr() that its *return* value must be used to access header fields after its call? This code however assumes that the header does span fragments and will be copied to `icmp_hdr` - I don't see why this would be true for this function.